code/modules/vehicles/mecha/mech_fabricator.dm

```diff
@@ -73,6 +73,8 @@
 	rmat = AddComponent(/datum/component/remote_materials, "mechfab", mapload && link_on_init, mat_container_flags=BREAKDOWN_FLAGS_LATHE)
 	RefreshParts() //Recalculating local material sizes if the fab isn't linked
 	update_menu_tech()
+	RegisterSignal(stored_research, COMSIG_TECHWEB_ADD_DESIGN, .proc/update_menu_tech)
+	RegisterSignal(stored_research, COMSIG_TECHWEB_REMOVE_DESIGN, .proc/update_menu_tech)
```

Comment:
Does this have signal handler?

Also for both of these just `RegisterSignal(x, list(signal1, signal2), .proc/update)`